HiP-EventStoreLib/EventStoreLlp/EventDataExtensions.cs

```diff
@@ -38,22 +54,61 @@ public static IEvent ToIEvent(this RecordedEvent ev)
                 throw new ArgumentException("The event is not JSON-formatted");
 
             var metadata = Encoding.UTF8.GetString(ev.Metadata);
-            var headers = JsonConvert.DeserializeObject<IDictionary<string, object>>(metadata);
+            var headers = JsonConvert.DeserializeObject<IDictionary<string, object>>(metadata, HeaderSerializationSettings);
 
             if (headers == null || !headers.TryGetValue(EventClrTypeHeader, out var typeNameEntry))
                 throw new ArgumentException("Cannot resolve event type: Metadata does not specify the CLR type");
 
             var typeName = typeNameEntry.ToString();
-            var type = Type.GetType(typeName);
+            var type = FindType(typeName);
 
             if (type == null)
                 throw new ArgumentException($"Cannot resolve event type: '{typeName}'");
 
             if (!typeof(IEvent).IsAssignableFrom(type))
                 throw new ArgumentException($"The event type does not implement {nameof(IEvent)}");
 
+            if (type == typeof(PropertyChangedEvent))
```

Comment:
What if `type` is a type derived from `PropertyChangedEvent`? Maybe we should seal the class `PropertyChangedEvent`.